qiskit_nature/mappers/second_quantization/logarithmic_mapper.py

```diff
@@ -0,0 +1,199 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Logarithmic Mapper."""
+
+import operator
+
+from fractions import Fraction
+from functools import reduce
+from typing import List, Union
+
+import numpy as np
+
+from qiskit.opflow import PauliSumOp
+from qiskit.quantum_info.operators import SparsePauliOp, Operator
+from qiskit_nature.operators.second_quantization import SpinOp
+from .spin_mapper import SpinMapper
+
+
+class LogarithmicMapper(SpinMapper):
+    """A mapper for Logarithmic spin-to-qubit mapping [1].
+
+
+    [1]: Mathis, S. V., Gauglielmo, M., & Ivano, T. (2020).
+         Toward scalable simulations of Lattice Gauge Theories on quantum computer.
+         `arXiv:2005.10271 <https://arxiv.org/abs/2005.10271>`_
+    """
+
+    def __init__(self, embed_padding: float = 1, embed_location: str = "upper") -> None:
+        """
+        Args:
+            embed_padding:
+                The value of the diagonal elements of the upper left  block of the embedded matrix.
+            embed_location:
+                Must be one of ['upper', 'lower']. This parameters sets whether
```

Comment:
If this can only take a limited set of strings I suggest we extract these into an `Enum`